server/src/test/java/org/elasticsearch/action/admin/cluster/configuration/TransportAddVotingConfigExclusionsActionTests.java

```diff
@@ -170,11 +168,11 @@ public void testWithdrawsVoteFromANode() throws InterruptedException {
             })
         );
 
-        assertTrue(countDownLatch.await(30, TimeUnit.SECONDS));
+        safeAwait(countDownLatch);
         assertThat(clusterService.getClusterApplierService().state().getVotingConfigExclusions(), contains(otherNode1Exclusion));
```

Comment:
You've moved some of the assertions about the current state into the success handler - I would slightly prefer them all to move, but whichever way we go I'd rather we were consistent.

Reply:
This comment is still outstanding.

Reply:
Apologies, I had just taken a quick look at the merge conflicts, I hadn't meant to give impression I had addressed the issues you raised. I should have made my intentions more explicit.

I was keeping some of the assertions outside the handler is because I'd, superficially, made the assumption the handler must be being called twice in order to decrement the `CountDownLatch` twice, so the assertion would need to be outside the handler to prevent it being called twice also. I can see this isn't the case now, so have moved the remaining assertions as well.